mesonbuild/build.py

```diff
@@ -857,6 +858,8 @@ def get_extra_args(self, language):
         return self.extra_args.get(language, [])
 
     def get_dependencies(self, exclude=None, internal=True):
```

Comment:
`@functools.lru_cache()` instead of manually managing the cache?

Reply:
Note that you will want to also convert the return value of this function from a mutable list to an immutable tuple else if someone accidentally edits the dependencies list it will change the cache.